Add Air Source Integrated Heat Pump and Fix Variable Speed Heat Pump Water Heater COP Bug

mbadams5 opened the pull request:
## Pull request overview

This pull request adds a new feature, Air Source Integrated Heat Pump, developed by Bo Shen at ORNL (@shenb). 
Also addresses variable speed heat pump water heater COP bug #5819.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [X] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - NewFeature: This pull request includes code to add a new feature to EnergyPlus
- [x] Develop alternate coil node name scheme (@mjwitte)
- [ ] Clean up new example file (remove tabs, use proper field names, indentation, placement of objects)
- [ ] Documentation
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] ExpandObjects changes??
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

shenb committed:
add latest IHP files

shenb committed:
check in changes for VS coils

shenb committed:
add changes of water tank

shenb committed:
add changes in Furnace

shenb committed:
change general routines to allow node connections in IHP

shenb committed:
add one example file

add ASIHPMixtank.idf to test files, and delete member construction
function in IHP

shenb committed:
specify SCWHmatchWH air flow rate

shenb committed:
save temp air flow rates

shenb committed:
check in idd

shenb committed:
update cmakelists.txt to include IHP cc and hh files

shenb committed:
add example file to cmak list in the folder

shenb committed:
add output variables and reverse back test files cmake list

shenb committed:
add back example file to cmake list

mbadams5 committed:
Merge branch 'develop' into ASIHPdev3

# Conflicts:
#	src/EnergyPlus/Furnaces.cc

mbadams5 committed:
Fix merge conflicts

# Conflicts:
#	src/EnergyPlus/Furnaces.cc

mbadams5 committed:
Change namespace, add enum class, reformat

mbadams5 committed:
Add unit tests

shenb committed:
fixed bugs using latest version

1. add a "Optional_bool_const SuppressWarning" to choose whether
printing the severe error in GetVSCoilIndex (existing implementation in
DX coils)
2. add a VSName string in the calc stand rating routine

mbadams5 committed:
Change how severe error message is handled

I also removed many unnecessary includes in header files

mbadams5 committed:
Fix integration failure

mbadams5 committed:
Change HasIHP function name

mbadams5 committed:
Fix warnings and integration failures

nrel-bot-3 commented:
@Myoldmopar @mbadams5 @lgentile it has been 14 days since this pull request was last updated.

Myoldmopar commented:
FYI @mbadams5 this is on my radar.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -42533,10 +42533,8 @@ Coil:Cooling:DX:VariableSpeed,
         \type alpha
         \reference CoolingCoilsDXVariableSpeed
    A2,  \field Indoor Air Inlet Node Name
-        \required-field
```
Hmm, implications of removing required-field for all variable speed DX coil objects?  Maybe this will be resolved as I make my way through the rest of the PR review, just spewing stream of consciousness as I go.  

FYI @rraustad @mjwitte @EnergyArchmage

rraustad replied:
The child needs node names, the parent does not since they can be mined from the child.

shenb replied:
To Edwin: 
Air source integrated heat pump (ASIHP) includes multiple, variable-speed coils, which have the same inlet and output nodes. Our design aims to have the ASIHP object to contain the nodes, and pass them into the “child” coil objects, in order to avoid repeating the nodes in the coils objects, which are prevented by the EP general logic now. So, we removed required-field of nodes for all variable speed DX coil objects. 

To Rich:
Air source integrated heat pump (ASIHP) includes multiple, variable-speed coils, which have the same inlet and output nodes. Our design aims to have the ASIHP object to contain the nodes, and pass them into the “child” coil objects, in order to avoid repeating the nodes in the coils objects, which are prevented by the EP general logic now. So, we removed required-field of nodes for all variable speed DX coil objects.

shenb replied:
Air source integrated heat pump (ASIHP) includes multiple, variable-speed coils, which have the same inlet and output nodes. Our design aims to have the ASIHP object to contain the nodes, and pass them into the “child” coil objects, in order to avoid repeatign the nodes in the coils objects, which are prevented by the EP general logic now. So, we removed required-field of nodes for all variable speed DX coil objects.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
         \note cubic curve = a + b*ffa + c*ffa**2 + d*ffa**3
         \note ffa = Fraction of the full load Air Flow
+						
+COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE,
```
We don't do all caps on object names, and there is a tab on the following \memo line

shenb replied:
fixed.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
+       \type alpha
+       \reference IntegratedHeatPumps
+       \note Unique name for this instance of an air-source integrated heat pump.
+  A2, \field Indoor Air Cooling Coil Inlet Node Name
```
Do you need Cooling Coil in this field name?  On this one you have:

`Indoor Air Cooling Coil Inlet Node Name`

But then below you have:

`Outdoor Air Inlet Node`

I would unify these to the minimum length that still fully defines the input.  If you can get away with just `Indoor Air Inlet Node Name` that's great.  And on that note, the below node inputs should all have a trailing "Name", right @mjwitte?  Well, whatever, they should be unified to the rest of the IDD.

shenb replied:
defining “cooling coil” “heating coil” and “outdoor” are necessary, as the ASIHP contains multiple nodes, separately for cooling coils, heating coils, and water heating coils with outdoor air node. ASIHP will pass these nodes to “child” coils accordingly. And defining separate “cooling coil” “heating coil” nodes is needed to accommodate the inputs of AirLoopHVAC:UnitaryHeatPump:AirToAir (a related object). We tried to shorten them, by deleting “indoor”, and add “Name” to all endings.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
+  A6, \field Condenser Water Outlet Node
+       \required-field
+       \type alpha
+  A7, \field Water Node to Monitor the Supply Hot Water Flow
```
That feels like an odd input field name.  Maybe `Supply Hot Water Flow Sensing Node Name` or something of that nature?

mjwitte replied:
Yes, but "Sensor" instead of "Sensing"

mjwitte replied:
All node names should be \type node

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
+  A9, \field Outdoor Air Outlet Node
+       \required-field
+       \type alpha
+  A10, \ field Coil Name of Space Cooling Mode
```
Space between slash and field won't be processed correctly by IP.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
+       \type object-list
+       \object-list CoolingCoilsDXVariableSpeed
+       \note Must match the name used in the corresponding Coil:Cooling:DX:VariableSpeed object.
+  A11,  \field Coil Name of Space Heating Mode
```
Yeah, all these passive voice input field names feel odd.  I'm curious @mjwitte's thoughts across the board, but I would think they would be better rearranged like:

```
Coil Name of Space Heating Mode
```

to:

```
Space Heating Coil Name
```

mjwitte replied:
Correct.  "Name" should generally be the last word of  a field name.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
+  A16, \field Heating Coil Name of Combined Space Heating and Water Heating Mode with Desuperheating
+       \type object-list
+       \object-list HeatingCoilsDXVariableSpeed
+	   \note Must match the name used in the corresponding Coil:Heating:DX:VariableSpeed object.   
```
Another stray \t

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
+       \type object-list
+       \object-list HeatPumpWaterHeaterDXCoils
+       \note Must match the name used in the corresponding Coil:WaterHeating:AirToWaterHeatPump:VariableSpeed object.	   	   
+  N1 , \field Indoor Temperature Limit for SCWH Mode
```
Acronym in the field name...I get that there is a balance between enormously long field names and ensuring the field name is fully understood standing alone.  I'd like to see these removed, and maybe there is a way to re-word it so that it isn't super long.

mjwitte replied:
These are difficult.  If I remember correctly, I pushed back on really long names in the original NFP.  I will wait for a set of revised field names throughout and then review them.  Words like "of" and "is", "to", "above which" should be gone from most field names, if not all.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
+  N2 , \field Ambient Temperature Limit for SCWH Mode
+       \type real
+       \units C
+       \minimum>20.0
```
I'm not sure if IP will have trouble with a missing space here, but I'd go ahead and add a space after the >

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45615,131 @@ Coil:Heating:DX:VariableSpeed,
+       \default 1
+       \minimum> 0
+       \maximum< 10
+  N9,  \field Maximum Running Time before Allowing Electric Res Heat Use during SHDWH Mode
```
See above comment regarding SHDWH, but for sure don't shorten Resistance to Res

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -52785,29 +52901,35 @@ AirLoopHVAC:UnitaryHeatPump:AirToAir,
         \type choice
         \key Coil:Heating:DX:SingleSpeed
         \key Coil:Heating:DX:VariableSpeed
+		\key COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE
```
Tabs, plus capitalization

Myoldmopar commented on src/EnergyPlus/BranchNodeConnections.cc:
```diff
@@ -665,6 +665,7 @@ namespace BranchNodeConnections {
 					}
 				}
 				if ( ! IsValid ) {
+					if (NodeConnections(Loop1).ObjectType == "COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE") continue;
```
Ugh.  Super-specialization for just this one component.  Looks uber-klugy.  Maybe it's the best path forward, as I see below the same sort of thing was done for the unglazed transpired solar collector, but it certainly doesn't _look_ great.  Continuing review...

mjwitte replied:
The purpose of compsets is to validate parent-child object connections.  It has two main functions:
1.  To catch cases where the parent object uses node names that don't match the child object.  e.g. The Branch object refers to a chiller with Inlet Node X and Outlet Node Y but the chiller has no nodes by those names.
2.  To catch cases where the same object is used in more than one place.  e.g. Chilled Water Coil 1 is used in two different fan coil units.  Normally, this is illegal.  But for transpired solar collectors and this critter CoilSystem:IntegratedHeatPump:AirSource, it _is_ legal for multiple parents to reference the same child.  In the example file for this, CoilSystem:IntegratedHeatPump:AirSource "ASIHP1" is referenced by a WaterHeater:HeatPump:PumpedCondenser, and an AirLoopHVAC:UnitaryHeatPump:AirToAir (twice).  So we need a general approach to allow components like this.  Let me think this through and offer a solution.

shenb replied:
if we didn’t bypass this, it would cause fatal error and never allow a run-through. You see the ASIHP violates some “fundamental” E+ rules, i.e. same air and water nodes are connected to many different coil objects, and more than one parent objects refer to the same ASIHP object, since ASIHP is a multi-functional object for space conditioning and water heating. Probably, bypassing these checks is an easier way to implement the ASIHP function without breaking too many E+ present logics and causing a major modification.

shenb replied:
[ResponseToMike.docx](https://github.com/NREL/EnergyPlus/files/301387/ResponseToMike.docx)

Myoldmopar commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -277,9 +279,8 @@ namespace Furnaces {
-
 	void
-	clear_state()
+		clear_state()
```
`'\t' > /dev/null`

shenb replied:
fixed.

Myoldmopar commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -480,6 +485,8 @@ namespace Furnaces {
 			if ( Furnace( FurnaceNum ).CoolingCoilType_Num == Coil_CoolingAirToAirVariableSpeed ) {
 				// variable speed cooling coil
 				HeatCoilLoad = 0.0;
+				if (true == Furnace(FurnaceNum).bIsIHP) IntegratedHeatPumps(Furnace(FurnaceNum).CoolingCoilIndex).ControlledZoneTemp =
```
`true == Furnace(FurnaceNum).bIsIHP`

No.

shenb replied:
all fixed.

Myoldmopar commented on src/EnergyPlus/BranchNodeConnections.cc:
```diff
@@ -1868,6 +1870,11 @@ namespace BranchNodeConnections {
 				//  All other values must match
 				AlreadyNoted( Other ) = true;
+
+				if ("COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE" == CompSets(Count).CType)	{
```
Until we have a better code style guide, new code should be uniform with the existing codebase, so you need to add padding within brackets and parentheses.  This comment should be applied throughout the code changes on this PR.

shenb replied:
done.

Myoldmopar commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -1940,7 +1975,14 @@ namespace Furnaces {
 					errFlag = false;
 					if ( Furnace( FurnaceNum ).CoolingCoilType_Num == Coil_CoolingAirToAirVariableSpeed ) {
-						Furnace( FurnaceNum ).CondenserNodeNum = GetVSCoilCondenserInletNode( CoolingCoilName, errFlag );
+						if (true == Furnace(FurnaceNum).bIsIHP) {
```
No.

shenb replied:
fixed.

Myoldmopar commented on src/EnergyPlus/SimulationManager.cc:
```diff
@@ -1951,6 +1951,7 @@ namespace SimulationManager {
 					ShowWarningError( "Node Connection errors shown during \"fatal error\" processing may be false because not all inputs may have been retrieved." );
 					WarningOut = false;
 				}
+				if (CompSets(Count).CType == "COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE") continue;
```
More specialization.  Feels like square peg in round hole.  Should we be restructuring at a different level to avoid these?  What is it about these units that is causing you to simply skip this error handling?

shenb replied:
We have to skip this error handling because ASIHP has a different node connection style, which cause fatal error here.

Myoldmopar commented on src/EnergyPlus/VariableSpeedCoils.cc:
```diff
@@ -242,6 +246,168 @@ namespace VariableSpeedCoils {
 		VarSpeedCoil.deallocate();
 	}
 
+	// Default Constructor
```
Hmmmmm default constructor in the cc file.  There's nothing wrong with this.  However, in the entirety of the code base we have defined the default constructors in the header in the struct definition.  Since we don't have a policy, could you match the rest of the codebase and simply put this over in the struct itself.  Shouldn't be anything but a cut/paste and removing the `VariableSpeedCoilData::` scope declaration.

shenb replied:
the default constructor is put here due to needing some pre-defined constant for initialization, like int const MaxSpedLevels( 10 ); // Maximum number of speed that supports
This will make it easy to increase the upper speed level later.

Myoldmopar replied:
What in the E+ source is causing the maximum speed levels to be 10?

mbadams5 replied:
[It is here](https://github.com/NREL/EnergyPlus/blob/fed09fcb8f94a93050a35c40b18bfb3f87b2e383/src/EnergyPlus/VariableSpeedCoils.cc#L164) but really I moved this because of `AutoSize` since I didn't want to `#include <DataHVACGlobals.hh>` or `#include <DataSizing.hh>` (I forget which) in the header file. I am trying to minimize the number of `#include` and `using` in header files.

Myoldmopar commented on src/EnergyPlus/VariableSpeedCoils.cc:
```diff
@@ -896,6 +1064,7 @@ namespace VariableSpeedCoils {
 			VarSpeedCoil( DXCoilNum ).AirInletNodeNum = GetOnlySingleNode( AlphArray( 2 ), ErrorsFound, CurrentModuleObject, AlphArray( 1 ), NodeType_Air, NodeConnectionType_Inlet, 1, ObjectIsNotParent );
 			VarSpeedCoil( DXCoilNum ).AirOutletNodeNum = GetOnlySingleNode( AlphArray( 3 ), ErrorsFound, CurrentModuleObject, AlphArray( 1 ), NodeType_Air, NodeConnectionType_Outlet, 1, ObjectIsNotParent );
 
+			if ((VarSpeedCoil(DXCoilNum).AirInletNodeNum > 0) && (VarSpeedCoil(DXCoilNum).AirOutletNodeNum > 0)) // IHP allows no node inputs for coil
```
So, this will skip the next line only?  What if you have a non IHP object?  Let's say the user specifies an invalid set of nodes.  They will both come back greater than zero.  So then it will skip the TestCompSet call, right?  This feels wrong.

mjwitte replied:
The var speed DX coil must have node names - making them optional is not good.  This is related to but not the same as the compsets issue - using the same coilsystem object to attach multiple coils in the same place in the system.  I'll propose a solution once I wrap my head around how these all connect together.

shenb replied:
Add a flag IsCallbyIHP in the function calling argument list to indicate if the initialization call is from an IHP object. 

i.e. change
if ((VarSpeedCoil(DXCoilNum).AirInletNodeNum > 0) && (VarSpeedCoil(DXCoilNum).AirOutletNodeNum > 0))  TestCompSet( CurrentModuleObject, AlphArray( 1 ), AlphArray( 2 ), AlphArray( 3 ), "Air Nodes" );
to 
if (!IsCallbyIHP) // IHP allows no node inputs for coil
            TestCompSet( CurrentModuleObject, AlphArray( 1 ), AlphArray( 2 ), AlphArray( 3 ), "Air Nodes" );

Myoldmopar commented on src/EnergyPlus/VariableSpeedCoils.cc:
```diff
@@ -4600,6 +4797,16 @@ namespace VariableSpeedCoils {
+		//Check for flows, do not perform simulation if no flow in load side or source side.
+		if ((SourceSideMassFlowRate <= 0.0) || (LoadSideMassFlowRate <= 0.0)) {
+			VarSpeedCoil(DXCoilNum).SimFlag = false;
+			return;
```
There wasn't an early `return` here before.  Is there anything that needs to be cleaned up before exiting this function?

shenb replied:
Nothing needs to be clean up here, as no node data and other “communicative” variables is modified up to this point.

Myoldmopar commented on src/EnergyPlus/VariableSpeedCoils.cc:
```diff
@@ -5834,6 +6069,124 @@ namespace VariableSpeedCoils {
 
 	}
 
+	void SetAirNodes(std::string const & CoilName, // must match coil names for the coil type
```
Recommended cleanups for this function.  Change declaration style (again, until we have a better style defined) to:

```
void SetAirNodes(
        std::string const & CoilName,
        etc...
) {
        code
}
```

Remove all the unnecessary header blocks ( REFERENCES, INTERFACE BLOCK SPECIFICATIONS, etc. )

Remove commented out code that isn't useful: 
`//    WaterIndex=FindGlycol('WATER') !Initialize the WaterIndex once`

Myoldmopar commented on src/EnergyPlus/VariableSpeedCoils.cc:
```diff
@@ -5834,6 +6069,124 @@ namespace VariableSpeedCoils {
+		}
+	}
+
+	void SetWaterNodes(std::string const & CoilName, // must match coil names for the coil type
```
Same comment as for previous function.

Myoldmopar commented on src/EnergyPlus/VariableSpeedCoils.cc:
```diff
@@ -6589,7 +6950,7 @@ namespace VariableSpeedCoils {
 			CBFAdj = std::exp( ADiff );
 		} else {
-			CBFAdj = 0.0;
+			CBFAdj = 1.0e-6;
```
Does this affect non-IHP objects?

shenb replied:
this voids data overflow. IHP and non-IHP objects need the same handling.

Myoldmopar commented on src/EnergyPlus/VariableSpeedCoils.hh:
```diff
@@ -370,168 +365,11 @@ namespace VariableSpeedCoils {
-			TotalHeatingEnergyRate( 0.0 ) //total WH energy rate
-			//end variables for HPWH
-		{}
+		VariableSpeedCoilData();
```
So there may have been a good reason to move the constructor.  As I mentioned above, it's not _necessarily_ wrong, just different from the rest of the codebase.  I'm fine having it moved if there is a reason.

shenb replied:
the default constructor is in C++ than the struct header file,  due to needing some pre-defined constant for initialization, like int const MaxSpedLevels( 10 ); // Maximum number of speed that supports
This will make it easy to increase the upper speed level later.

mjwitte replied:
Then the header file needs comments here to explain why this is different.

mbadams5 replied:
@mjwitte @Myoldmopar I was the one who made this change. I did this so I could remove all the `#include`s from this header file as well as the `using` statements. Since the initializer list uses different variables from other header files, this is the only way to remove those include statements (since they then move to the cc file as well). Reducing `#include` statements helps to reduce compile time, header pollution, and rebuild time if one of those header files is altered.

mjwitte replied:
@mbadams5 That's fine, I just think it warrants a brief comment in this header file to explain why it's different.

Myoldmopar commented on src/EnergyPlus/WaterThermalTanks.cc:
```diff
@@ -56,19 +56,17 @@
 
+
 // C++ Headers
-#include <cassert>
```
Removing unused?  If so, awesome!

shenb replied:
[ResponseToEdwin.docx](https://github.com/NREL/EnergyPlus/files/301381/ResponseToEdwin.docx)

shenb replied:
didn't find this line
-#include <cassert>
in WaterThermalTanks.cc, probably, it has been updated by Mark.

shenb replied:
[ResponseToEdwin.docx](https://github.com/NREL/EnergyPlus/files/301399/ResponseToEdwin.docx)

Myoldmopar commented:
So I made many comments.  It would be awesome if you could take those comments and propagate them through the entire PR.  I didn't make a comment everywhere there were whitespace problems or checking `if (true == something)`, so just look over the entire set of changes and clean them all up.  Once that is complete, and my other functional questions have been resolved, I'll make another pass.

@rraustad would you mind taking a peek at the changes to [Furnaces](https://github.com/NREL/EnergyPlus/pull/5584/files#diff-5d2c4340e747a9c0a89e7109e2ff99a1)?

rraustad commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -866,6 +885,8 @@ namespace Furnaces {
 		Real64 SteamDensity; // density of steam at 100C
 		int DXCoilIndex; // Index to DX coil in HXAssited object
+		std::string IHPCoilName;//IHP cooling coil name
+		int IHPCoilIndex(0);//IHP cooling coil id
```
These new locals are confusing since they refer to different coils at different places in GetInput. Why not just use existing locals CoolingCoilName and HeatingCoilName? or if a new coil type is needed then be explicit like using CondenserCoilName.

shenb replied:
better to start a new IHPCoilName and IHPCoilIndex, since IHP is a collection of coils, than a single cooling or heating coil name.

rraustad commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -921,6 +942,11 @@ namespace Furnaces {
 		CheckEquipName.dimension( NumFurnaces, true );
 
+		for (IHPCoilIndex = 1; IHPCoilIndex <= NumFurnaces; ++IHPCoilIndex)//clear IHP flag
+			Furnace(IHPCoilIndex).bIsIHP = false;
```
Is this needed? bIsIHP is already set to false in the header so no need to do it again.

rraustad commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -8707,6 +8930,8 @@ namespace Furnaces {
 			PartLoadFrac = 0.0;
 		}
 
+		if (true == Furnace(FurnaceNum).bIsIHP) SpeedNum = GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex);
```
Don't you already know the max speed for each coil? Isn't it SpeedNum = Furnace(FurnaceNum).NumOfSpeed. If not, you should save this during GetInput so you don't have to call this function every time.

rraustad commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -9445,7 +9716,59 @@ namespace Furnaces {
+					CompOffFlowRatio = CompOffMassFlow /
+						GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex), 1.0, false);
+				}
+			}
```
Lot's of GetAirMassFlowRateIHP calls here. Should have an array of Mdot vs speed num to avoid excess function calls. Did you look at SetOnOffMassFlowRate or SetOnOffMassFlowRateVSCoil to see that MSHPassFlowRateLow/High are set there automatically?

rraustad replied:
It looks like you added your own SetOnOffMassFlowRate function, that's a good thing. I can't tell from this GitHub view where these lines of code are wrt existing.

shenb replied:
the max speed will differ by individual working modes. So, GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex) will check the current work mode first, and then, get the max speed. 

the air flow rate will differ by individual working modes (more than 5 modes) and speed levels. So, GetAirMassFlowRateIHP will check the current work mode and allowed speed first, and then, get the air flow rate. They can’t be contained in one array.

rraustad commented on src/EnergyPlus/Furnaces.hh:
```diff
@@ -442,7 +445,7 @@ namespace Furnaces {
+		clear_state();
 
 	void
 	SimFurnace(
```
Reviewed all the way through Furnaces. Things look relatively similar to existing code.

shenb replied:
many thanks for your effort.

shenb replied:
[ResponseToRichard.docx](https://github.com/NREL/EnergyPlus/files/301386/ResponseToRichard.docx)

nrel-bot-3 commented:
@Myoldmopar @mbadams5 @lgentile it has been 14 days since this pull request was last updated.

shenb committed:
changes according to reviewers' comments

respond to Edwin, Mike and Richard comments after the PR review

shenb committed:
avoid node connection warnings

indicate calling from IHP when call GetVSCoilIndex and GetVSCoilInput()

shenb committed:
delete one tab

mjwitte commented:
@mbadams5 @shenb There are two older branches named `ASIHPDev` and `ASIHPDev2`. Can these be deleted?

shenb commented:
Yes, they are obsolete and can be deleted.

Thanks,

Bo

From: Michael J. Witte [mailto:notifications@github.com]
Sent: Wednesday, June 15, 2016 10:29 AM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

@mbadams5https://github.com/mbadams5 @shenbhttps://github.com/shenb There are two older branches named ASIHPDev and ASIHPDev2. Can these be deleted?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-226204588, or mute the threadhttps://github.com/notifications/unsubscribe/AFrLMneFISLlB_eiA2w_Oj0VFPQJfCZAks5qMAwogaJpZM4H9ELN.

shenb committed:
set back required nodes to VS coils

set back nodes of VS coils as required fields; add IsCallbyIHP flag when
initializing the VS coils; if IsCallbyIHP =true, will skip node checking
in VS coils, and leave the checking to the IHP object; if IsCallbyIHP =
false (default), will proceed as a regular VS coil object

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -66632,15 +66741,17 @@ WaterHeater:HeatPump:PumpedCondenser,
        \type choice
        \key Coil:WaterHeating:AirToWaterHeatPump:Pumped
        \key Coil:WaterHeating:AirToWaterHeatPump:VariableSpeed
+	   \key COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE
```
Case is wrong here and two places below.

Myoldmopar replied:
And there are tabs/indent problems scattered in here.

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -45619,7 +45619,110 @@ Coil:Heating:DX:VariableSpeed,
+       \note Must match the name used in the corresponding Coil:Heating:DX:VariableSpeed object.
+  A5, \field Dedicated Water Heating Coil Name
+       \type object-list
+       \object-list HeatPumpWaterHeaterDXCoils
```
`HeatPumpWaterHeaterDXCoils` does not appear as `\reference HeatPumpWaterHeaterDXCoils` anywhere in the idd.  Need to add that to the appropriate objects.

mjwitte replied:
Or change these object-list to use existing reference name(s).  You can have more than one object-list on the same field.

mjwitte committed:
ASIHP resolve node connection problems

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into ASIHPdev3

Conflicts:
	src/EnergyPlus/Furnaces.cc

mjwitte committed:
ASIHP resolve node connection problems - cleanup

mjwitte commented:
@shenb My changes broke one of the unit tests.  Let me know if you need help fixing that.  Also, VSHeatPumpWaterHeater regression tests shows large diffs.  That seems wrong.

shenb commented:
Mike,

VSHeatPumpWaterHeater regression tests shows large diffs --> this is expected, as we just fixed a bug in the related codes.

We will look at the unit test.

Thanks,

Bo

From: Michael J. Witte [mailto:notifications@github.com]
Sent: Tuesday, June 21, 2016 2:18 PM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

@shenbhttps://github.com/shenb My changes broke one of the unit tests. Let me know if you need help fixing that. Also, VSHeatPumpWaterHeater regression tests shows large diffs. That seems wrong.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-227526353, or mute the threadhttps://github.com/notifications/unsubscribe/AFrLMvP4CTzjN2fxEYYmeKKccTDE6YfUks5qOCr0gaJpZM4H9ELN.

mjwitte commented:
Then we need to document the bug as an issue and reference that it is fixed here.

shenb commented:
Sure, will do.

From: Michael J. Witte [mailto:notifications@github.com]
Sent: Tuesday, June 21, 2016 2:37 PM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

Then we need to document the bug as an issue and reference that it is fixed here.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-227531578, or mute the threadhttps://github.com/notifications/unsubscribe/AFrLMoFdDv7mV-1GkTuGmxYxDUfKG3Boks5qOC80gaJpZM4H9ELN.

shenb committed:
fix unit test

update unit test cc to accomodate IDD change

shenb commented:
Mike,

Your improvement on the ASIHP looks excellent to me.

We’ve fixed the unit test, and pushed the latest update to ASIHPdev3.

Thanks,

Bo

From: Michael J. Witte [mailto:notifications@github.com]
Sent: Tuesday, June 21, 2016 2:18 PM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

@shenbhttps://github.com/shenb My changes broke one of the unit tests. Let me know if you need help fixing that. Also, VSHeatPumpWaterHeater regression tests shows large diffs. That seems wrong.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-227526353, or mute the threadhttps://github.com/notifications/unsubscribe/AFrLMvP4CTzjN2fxEYYmeKKccTDE6YfUks5qOCr0gaJpZM4H9ELN.

shenb committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into ASIHPdev3

Myoldmopar commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -8684,6 +8914,13 @@ namespace Furnaces {
+		{
+			IHPMode = GetCurWorkMode(Furnace(FurnaceNum).CoolingCoilIndex);
+			if ((IHPOperationMode::DWHMode == IHPMode) || (IHPOperationMode::SCWHMatchWHMode == IHPMode)) //cooling capacity is a resultant
+				return;
```
Please brace if blocks, even if just a single line, for readability.

Myoldmopar commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -8996,26 +9238,40 @@ namespace Furnaces {
 				SaveCompressorPLR = VarSpeedCoil( Furnace( FurnaceNum ).CoolingCoilIndex ).PartLoadRatio;
 			} else {
-				SimVariableSpeedCoils( BlankString, Furnace( FurnaceNum ).CoolingCoilIndex, Furnace( FurnaceNum ).OpMode, Furnace( FurnaceNum ).MaxONOFFCyclesperHour, Furnace( FurnaceNum ).HPTimeConstant, Furnace( FurnaceNum ).FanDelayTime, CompOp, 0.0, 1, 0.0, 0.0, 0.0, OnOffAirFlowRatio );
+				if(false == Furnace(FurnaceNum).bIsIHP)
```
⛔

Myoldmopar commented on src/EnergyPlus/IntegratedHeatPump.hh:
```diff
@@ -0,0 +1,413 @@
+
+	namespace IntegratedHeatPump {
+
+		// Using/Aliasing
```
Lots of empty comments, please clean up.

Myoldmopar commented:
I added a few more comments, please address those, clean up all the missing padding in the parentheses, and also add the LaTeX documentation, and then this should be very close to done.

mbadams5 committed:
Merge branch 'develop' into ASIHPdev3

mbadams5 committed:
Merge branch 'develop' into ASIHPdev3

mbadams5 committed:
Reformat code and address comments

mbadams5 commented:
@Myoldmopar I addressed your comments, cleaned up padding (actually automatically reformatted the files so you might need to use `?w=1` now to avoid whitespace diffing). Either Bo or myself will need to work on the LaTeX documentation.

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into ASIHPdev3

Myoldmopar committed:
Doing some cleanup; trying to resolve the huge add/delete diffs in Furnaces

Myoldmopar committed:
Clean up indent in Furnaces causing all diff lines

Myoldmopar committed:
Continue cleaning up indent problems

Myoldmopar committed:
Continued reverting indent and style things

Myoldmopar committed:
One more cleanup pass on Furnaces

Myoldmopar committed:
Clean up select case indents in var speed coils

Myoldmopar committed:
Clean up in new IHP code

Myoldmopar commented:
OK, so I made a bunch of cleanup passes, reviewing it along the way.  Could one of you address @mjwitte's remaining IDD comment about the object-list not being referenced anywhere.  After that, I'll start running some tests to move toward getting this merged in.

shenb commented:
Something seems broken in the recent merge. The main new example files now crashes. I will look into this.

Thanks,

Bo

Program Version,EnergyPlus, Version 8.6.0-Unknown, YMD=2016.08.01 12:06,IDD_Version 8.5.0
   *\* Warning *\* Version: in IDF="8.5" not the same as expected="8.6"
   *\* Severe  *\* GetObjectDefMaxArgs: Did not find object="AirflowNetwork:ZoneControl:PressureController" in list of objects.
   *\* Severe  *\* GetObjectDefMaxArgs: Did not find object="AirflowNetwork:Distribution:Component:OutdoorAirFlow" in list of objects.
   *\* Severe  *\* GetObjectDefMaxArgs: Did not find object="AirflowNetwork:Distribution:Component:ReliefAirFlow" in list of objects.
   *\* Severe  *\* WaterHeater:HeatPump:PumpedCondenser="OUTDOORHEATPUMPWATERHEATER":
   *\*   ~~~   *\* Heat pump water heater fan outlet node name does not match next connected component.
   *\*   ~~~   *\* Fan outlet node name = HPOUTDOORFANAIROUTLETNODE
   *\*   ~~~   *\* Expected fan outlet node name = DX COOLING COIL AIR INLET NODE
   *\* Severe  *\* GetCoilOutletNodeVariableSpeed: Could not find CoilType="COILSYSTEM:INTEGRATEDHEATPUMP:AIRSOURCE" with Name="ASIHP1"
   *\* Severe  *\* WaterHeater:HeatPump:PumpedCondenser="OUTDOORHEATPUMPWATERHEATER":
   *\*   ~~~   *\* Heat pump water heater coil outlet node name does not match next connected component.
   *\*   ~~~   *\* Expected coil outlet node name = HPOUTDOORAIROUTLETNODE
   *\*  Fatal  *\* GetWaterThermalTankInput: Errors found in getting thermal storage input. Preceding condition(s) causes termination.
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=6
   ..... Last severe error=WaterHeater:HeatPump:PumpedCondenser="OUTDOORHEATPUMPWATERHEATER":
   ************\* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************\* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************\* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************\* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
   ************\* EnergyPlus Terminated--Fatal Error Detected. 1 Warning; 6 Severe Errors; Elapsed Time=00hr 01min 45.08sec

From: Edwin Lee [mailto:notifications@github.com]
Sent: Friday, July 29, 2016 12:48 PM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

OK, so I made a bunch of cleanup passes, reviewing it along the way. Could one of you address @mjwittehttps://github.com/mjwitte's remaining IDD comment about the object-list not being referenced anywhere. After that, I'll start running some tests to move toward getting this merged in.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-236232296, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFrLMqAGHDL-6d5L10lAbR33k-FwDxN-ks5qai7CgaJpZM4H9ELN.

shenb committed:
fix bugs occur after latest merging

correct HPWH IHP coil air side node connections during getinput

shenb committed:
address Mike's comment

HeatPumpWaterHeaterDXCoils does not appear as \reference
HeatPumpWaterHeaterDXCoils anywhere in the idd. Need to add that to the
appropriate objects.

correct HeatPumpWaterHeaterDXCoils to HeatPumpWaterHeaterDXCoils
VariableSpeed, which has the reference in varaible HPWH coil

shenb committed:
change version of example file

shenb committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into ASIHPdev3

shenb commented:
Edwin,

Thanks for help reviewing and improving the codes.

I’ve added a few more check-ins to resolve new conflicts due to the recent merging, and addressed Mike’s comment about the IDD. And, I merged latest Develop to the ASIHP branch yesterday. We think the codes are ready for the next step.

In the regression testing, there is one big diff, which is expected, for we corrected a bug in the previous variable-speed coil codes when developing the new feature.

I will work on the LaTeX doc next. Have to say this is my first time using LaTeX, may spend a bit extra time going through the learning curve.

Best,

Bo

From: Edwin Lee [mailto:notifications@github.com]
Sent: Friday, July 29, 2016 12:48 PM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

OK, so I made a bunch of cleanup passes, reviewing it along the way. Could one of you address @mjwittehttps://github.com/mjwitte's remaining IDD comment about the object-list not being referenced anywhere. After that, I'll start running some tests to move toward getting this merged in.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-236232296, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFrLMqAGHDL-6d5L10lAbR33k-FwDxN-ks5qai7CgaJpZM4H9ELN.

mbadams5 committed:
Merge branch 'develop' into ASIHPdev3

# Conflicts:
#	src/EnergyPlus/WaterThermalTanks.cc
#	tst/EnergyPlus/unit/CMakeLists.txt
#	tst/EnergyPlus/unit/WaterThermalTanks.unit.cc

mbadams5 committed:
Fix merge conflicts

Myoldmopar commented:
@mbadams5 @shenb The diffs in VSHeatPumpWaterHeater are _big_.  Are you sure it's behaving properly now?  You talk about above how you were going to [open an issue](https://github.com/NREL/EnergyPlus/pull/5584#issuecomment-227531858) so that we could track it, but I don't [see any](https://github.com/NREL/EnergyPlus/issues/created_by/shenb).  This can go in because everything else looks good, and you seem confident the diffs are just the bug fix, but it would be great to see some justification.  I'll watch for the documentation commit.

shenb commented:
Edwin,

We are sure this is a bug fix. There was a “typo” where I reversed COP for calculating the HPWH efficiency. That would surely lead to a big diff.

Thanks for point out the link where we can open an issue. I will open the issue right away.

Regarding the documentation,

I downloaded MikTex 2.9 as suggested at EP website and tried to compile and generate a PDF file for a 300 hundred of lines of codes. But, it seems taking for ever to generate the PDF. Am wondering if you have some suggestions on this?

Thanks,

Bo

[cid:image001.png@01D1F8A4.72AE8520]

From: Edwin Lee [mailto:notifications@github.com]
Sent: Thursday, August 18, 2016 7:38 AM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

@mbadams5https://github.com/mbadams5 @shenbhttps://github.com/shenb The diffs in VSHeatPumpWaterHeater are big. Are you sure it's behaving properly now? You talk about above how you were going to open an issuehttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-227531858 so that we could track it, but I don't see anyhttps://github.com/NREL/EnergyPlus/issues/created_by/shenb. This can go in because everything else looks good, and you seem confident the diffs are just the bug fix, but it would be great to see some justification. I'll watch for the documentation commit.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-240698450, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFrLMseU_2hDSwdMCOqNkmfgWjl7DY5Pks5qhEQjgaJpZM4H9ELN.

mjwitte commented:
@shenb Local doc builds can take a while (10 minutes or so on my windows machine).  The CI system now builds docs, so you can push up your changes and review the results there, but that's not instant either.  See the documents link near the top of the [Linux regression results](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-0703f05af7c5a2c69b17931eca8a62d6ba0474ac-x86_64-Linux-Ubuntu-14.04-gcc-4.8.html) or use the page icon link on the [dashboard](https://nrel.github.io/EnergyPlusBuildResults/) for your branch.

shenb commented:
Thanks Mike. Guess I wasn’t patient enough waiting it to get through. I will try pushing it up to the server to see the build next.

From: Michael J. Witte [mailto:notifications@github.com]
Sent: Thursday, August 18, 2016 9:59 AM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

@shenbhttps://github.com/shenb Local doc builds can take a while (10 minutes or so on my windows machine). The CI system now builds docs, so you can push up your changes and review the results there, but that's not instant either. See the documents link near the top of the Linux regression resultshttps://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-0703f05af7c5a2c69b17931eca8a62d6ba0474ac-x86_64-Linux-Ubuntu-14.04-gcc-4.8.html or use the page icon link on the dashboardhttps://nrel.github.io/EnergyPlusBuildResults/ for your branch.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-240731282, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFrLMkX8X7xFM6GrmCwt7hHgosd_35Peks5qhGU5gaJpZM4H9ELN.

shenb commented:
Edwin,

I just opened an new issue,

Bug fix for variable speed heat pump water heater coil, COP calculation #5819

Regarding the diffs in VSHeatPumpWaterHeater.

Thanks,

Bo

From: Edwin Lee [mailto:notifications@github.com]
Sent: Thursday, August 18, 2016 7:38 AM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

@mbadams5https://github.com/mbadams5 @shenbhttps://github.com/shenb The diffs in VSHeatPumpWaterHeater are big. Are you sure it's behaving properly now? You talk about above how you were going to open an issuehttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-227531858 so that we could track it, but I don't see anyhttps://github.com/NREL/EnergyPlus/issues/created_by/shenb. This can go in because everything else looks good, and you seem confident the diffs are just the bug fix, but it would be great to see some justification. I'll watch for the documentation commit.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-240698450, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFrLMseU_2hDSwdMCOqNkmfgWjl7DY5Pks5qhEQjgaJpZM4H9ELN.

shenb commented:
Edwin,

I think we’ve finished the LaTeX document for the ASIHP engineering reference, and I/o Reference. Attached please see our LaTeX codes and also the resultant PDF files as a proof.

At present, we are not sure where to add them in the whole docs. Can you please tell where we’d put them in?

Thanks,

Bo

From: Edwin Lee [mailto:notifications@github.com]
Sent: Thursday, August 18, 2016 7:38 AM
To: NREL/EnergyPlus
Cc: Shen, Bo; Mention
Subject: Re: [NREL/EnergyPlus] Add Air Source Integrated Heat Pump (#5584)

@mbadams5https://github.com/mbadams5 @shenbhttps://github.com/shenb The diffs in VSHeatPumpWaterHeater are big. Are you sure it's behaving properly now? You talk about above how you were going to open an issuehttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-227531858 so that we could track it, but I don't see anyhttps://github.com/NREL/EnergyPlus/issues/created_by/shenb. This can go in because everything else looks good, and you seem confident the diffs are just the bug fix, but it would be great to see some justification. I'll watch for the documentation commit.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/NREL/EnergyPlus/pull/5584#issuecomment-240698450, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFrLMseU_2hDSwdMCOqNkmfgWjl7DY5Pks5qhEQjgaJpZM4H9ELN.

mjwitte commented:
@shenb Regarding placement in the docs - and in the idd - please move the new object in the idd to be after CoilSystem:Cooling:DX:HeatExchangerAssisted.  This groups the CoilSystem objects together.  

The I/O Ref section can also go after CoilSystem:Cooling:DX:HeatExchangerAssisted (actually the I/O Ref isn't in the same order here as the IDD - grrr, so put this one after CoilSystem:Cooling:Water:HeatExchangerAssisted) these are in file \doc\input-output-reference\src\overview\group-heating-and-cooling-coils.tex

In the Engineering Reference, add it just before the section "Single-Speed Electric Heat Pump DX Water Heating Coil" in \doc\engineering-reference\src\simulation-models-encyclopedic-reference-001\coils.tex

mbadams5 committed:
Merge branch 'develop' into ASIHPdev3

# Conflicts:
#	src/EnergyPlus/Furnaces.cc
#	src/EnergyPlus/WaterThermalTanks.cc

mbadams5 committed:
Fix merge conflicts

# Conflicts:
#	src/EnergyPlus/Furnaces.cc
#	src/EnergyPlus/WaterThermalTanks.cc

shenb committed:
add docs

@shenb Regarding placement in the docs - and in the idd - please move
the new object in the idd to be after
CoilSystem:Cooling:DX:HeatExchangerAssisted. This groups the CoilSystem
objects together.
The I/O Ref section can also go after
CoilSystem:Cooling:DX:HeatExchangerAssisted (actually the I/O Ref isn't
in the same order here as the IDD - grrr, so put this one after
CoilSystem:Cooling:Water:HeatExchangerAssisted) these are in file
\doc\input-output-reference\src\overview\group-heating-and-cooling-coils.tex
In the Engineering Reference, add it just before the section
"Single-Speed Electric Heat Pump DX Water Heating Coil" in
\doc\engineering-reference\src\simulation-models-encyclopedic-reference-001\coils.tex

Myoldmopar committed:
Use new Fuel Heating Coil in Example File

Myoldmopar commented:
OK, cleaned up one integration test (Coil:Heating:Fuel), so CI will have one last pass methinks.

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into ASIHPdev3

Myoldmopar commented:
OK, CI has had its start on the latest tip of develop; so this is going in now.
